backend/src/v2/compiler/tektoncompiler/dag.go

```diff
@@ -378,7 +378,7 @@ func (i *pubDagDriverInputs) getParentDagID(isExitHandler bool) string {
 	if isExitHandler && i.parentDagID == compiler.RootComponentName {
 		return fmt.Sprintf("$(params.%s)", paramParentDagID)
 	} else if i.inLoopDag {
-		return fmt.Sprintf("$(params.%s)", paramParentDagID)
+		return taskOutputParameter(getDAGDriverTaskName(i.parentDagID), paramExecutionID)
```

Comment:
merge this with the `else` block?

Reply:
for me, logically, merging them and adding comments there would be good. otherwise, it's confusing.